vrx518_tc: allow larger MTU to support RFC4638

aimacintyre opened the pull request:
vrx518_tc currently sets the interface maximum MTU to the ethernet default of 1500 bytes by default via ether_setup() called from ptm_setup().

To support 1508 byte baby jumbo frames (RFC4638) for PPPoE connections over VDSL links as already supported by the VR9 ltq_ptm driver ([1], [2]) set the interface maximum MTU to MAX_MTU.

MAX_MTU is defined in dcdp/inc/tc_common.h to 2002 bytes and this value is used in ptm_change_mtu() and elsewhere as the maximum MTU, however this is short circuited by checks against the interface maximum MTU.

[1]: https://forum.openwrt.org/t/fritzbox-7530-and-rfc4638-baby-jumbo-frames/181327
[2]: https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=8a2a20e71e2909f84dab47e51dfda9e292a6c1ae

Signed-off-by: Andrew MacIntyre <andymac@pcug.org.au>

mrkiko commented:
Applied this patch on a FRITZ!BOX 7530 with no problems so far.
The configuration is unchanged, MTU unaltered.

aimacintyre commented:
> The configuration is unchanged, MTU unaltered.

Sorry, I'm not sure with this comment whether you're reporting that the patch didn't change anything without explicit action to set the interface MTU to 1508 (which is the expected behaviour) or whether you expected it would change the default interface MTU to 1508 but reporting that this didn't happen?

To clarify: the patch just increases the maximum permitted MTU on the ptm interface but doesn't change the default MTU (1500); a user wanting to use RFC4638 frames would explicitly set the ptm interface MTU to 1508 (and in the likely usage, also set the associated PPPoE interface MTU to 1500 instead of the PPPoE default MTU of 1492).

mrkiko commented:
Hello! I did not have expectations - just wanted to say it doesn't break things in my setup where I did not change MTU.
It was just a confirmation that all seems fine, at least without mtu changes. I mean - an assurance on the fact this is not breaking current setups.

On Tue, Nov 05, 2024 at 03:42:48AM -0800, Andrew MacIntyre wrote:
> 
> &gt; The configuration is unchanged, MTU unaltered.
> 
> Sorry, I&#39;m not sure with this comment whether you&#39;re reporting that the patch didn&#39;t change anything without explicit action to set the interface MTU to 1508 (which is the expected behaviour) or whether you expected it would change the default interface MTU to 1508 but reporting that this didn&#39;t happen?
> 
> To clarify: the patch just increases the maximum permitted MTU on the ptm interface but doesn&#39;t change the default MTU (1500); a user wanting to use RFC4638 frames would explicitly set the ptm interface MTU to 1508 (and in the likely usage, also set the associated PPPoE interface MTU to 1500 instead of the PPPoE default MTU of 1492).
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/openwrt/openwrt/pull/16856#issuecomment-2456949459
> You are receiving this because you commented.
> 
> Message ID: ***@***.***&gt;

robimarko commented on package/kernel/lantiq/vrx518_tc/patches/206-dcdp-ptm_tc-allow-larger-mtu-to-support-rfc4638.patch:
Can you please export this a full git patch with a title and description?
It is now clear what was the intention but it will be lost to time.

aimacintyre replied:
Hope the revision is what you're looking for.

robimarko replied:
Yes, indeed

aimacintyre committed:
vrx518_tc: allow larger MTU to support RFC4638

vrx518_tc currently sets the interface maximum MTU to the ethernet default
of 1500 bytes by default via ether_setup() called from ptm_setup().

To support 1508 byte baby jumbo frames (RFC4638) for PPPoE connections
over VDSL links as already supported by the VR9 ltq_ptm driver ([1], [2])
set the interface maximum MTU to MAX_MTU.

MAX_MTU is defined in dcdp/inc/tc_common.h to 2002 bytes and this value is
used in ptm_change_mtu() and elsewhere as the maximum MTU, however this is
short circuited by checks against the interface maximum MTU.

[1]: https://forum.openwrt.org/t/fritzbox-7530-and-rfc4638-baby-jumbo-frames/181327
[2]: https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=8a2a20e71e2909f84dab47e51dfda9e292a6c1ae

Signed-off-by: Andrew MacIntyre <andymac@pcug.org.au>
Link: https://github.com/openwrt/openwrt/pull/16856
Signed-off-by: Robert Marko <robimarko@gmail.com>
